COMPATIBILITY.md

```diff
@@ -0,0 +1,102 @@
+# `rules_rust` backwards compatibility policy
+
+This document defines the backwards compatibility policy for rules\_rust and
+defines the process for making compatibility breaking changes. Any exception to
+this process will have to be thoroughly discussed and there will have to be a
+consensus between maintainers that the exception should be granted.
+
+## What is a compatibility breaking change?
+
+A backwards compatible change has the property that a build that is green,
+correct, is using stable APIs of `rules_rust,` and is using a supported version
+of Bazel before the change is still green and correct after the change. In
+practice it means that users should not have to modify their project source
+files, their `BUILD` files, their `bzl` files, their `rust_toolchain`
+definitions, their platform definitions, or their build settings.
+
+A backwards compatible release has the property that it only contains backward
+compatible changes.
+
+`rules_rust` follow [SemVer 2.0.0](https://semver.org/). `rules_rust` promise
```

Comment:
> `rules_rust` promise that all minor version number and patch version number releases only contain backwards compatible changes.

Does this also apply to sub `1.0.0` releases? Is the plan to release a `1.0.0`?

Reply:
Oh good point, I should talk about pre-1.0 policy. In short I would make 0.X releases backwards incompatible, I'd ask breaking changes to follow the process, but maybe have shorter minimum time for migration than 3 months. WDYT?

Reply:
Yeah, until there's some concurrence on the current API in its entirety being stable, I'd say `0.X` releases are backwards incompatible and should have a short migration period. I'd love to get to a `1.0` ASAP and to do so I think we need the ability to relatively rapidly make changes. A note about the pre-`1.0` phase would be good.

Reply:
Added a section, please take a look :)